server/server.go

```diff
@@ -86,39 +88,59 @@ func New(cfg Config) (*Server, error) {
 		return nil, err
 	}
 
+	// Prometheus histograms for requests.
 	requestDuration := prometheus.NewHistogramVec(prometheus.HistogramOpts{
 		Namespace: cfg.MetricsNamespace,
 		Name:      "request_duration_seconds",
 		Help:      "Time (in seconds) spent serving HTTP requests.",
 		Buckets:   prometheus.DefBuckets,
 	}, []string{"method", "route", "status_code", "ws"})
-
 	prometheus.MustRegister(requestDuration)
 
-	router := mux.NewRouter()
-	router.Handle("/metrics", prometheus.Handler())
-	router.Handle("/traces", loki.Handler())
-	router.PathPrefix("/debug/pprof").Handler(http.DefaultServeMux)
-
+	// Setup gRPC server
 	grpcMiddleware := []grpc.UnaryServerInterceptor{
 		middleware.ServerLoggingInterceptor(cfg.LogSuccess),
 		middleware.ServerInstrumentInterceptor(requestDuration),
 		otgrpc.OpenTracingServerInterceptor(opentracing.GlobalTracer()),
 	}
 	grpcMiddleware = append(grpcMiddleware, cfg.GRPCMiddleware...)
-
 	grpcServer := grpc.NewServer(
 		grpc.UnaryInterceptor(grpc_middleware.ChainUnaryServer(
 			grpcMiddleware...,
 		)),
 	)
 
+	// Setup HTTP server
+	router := mux.NewRouter()
+	router.Handle("/metrics", prometheus.Handler())
+	router.Handle("/traces", loki.Handler())
+	router.PathPrefix("/debug/pprof").Handler(http.DefaultServeMux)
+	httpMiddleware := []middleware.Interface{
+		middleware.Log{
+			LogSuccess: cfg.LogSuccess,
+		},
+		middleware.Instrument{
+			Duration:     requestDuration,
+			RouteMatcher: router,
+		},
+		middleware.Func(func(handler http.Handler) http.Handler {
+			return nethttp.Middleware(opentracing.GlobalTracer(), handler)
+		}),
+	}
+	httpMiddleware = append(httpMiddleware, cfg.HTTPMiddleware...)
```

Comment:
No no this isn't going to work, not all the http paths need the same middleware (ie auth)